Acquisition/Poll/source/mainmonitor.cpp

```diff
@@ -0,0 +1,271 @@
+/** \file mainmonitor.cpp
+ *
+ * \brief Receives and decodes rate packets from StatsHandler, and rebroadcasts packets to the submonitors
+ *
+ * \author T.T. King, Cory R. Thornsberry, K. Smith, and S. Burcher
+ *
+ * \date Feb 19, 2023
+ *
+ */
+
+#include "mainmonitor.hpp"
+using namespace std;
+
+// TODO test what happens when subs are ctrlC'd first then try to quit main
+// TODO also figureout best way to handle reading in the Max_Num_submontor var
+//  Define the function to be called when ctrl-c (SIGINT) is sent to process
+void signal_callback_handler(int signum) {
+    cout << "\n    SIGINT:: Sending KILL_SOCKET to the submonitors\n\n"
+         << endl;
+    if (!DumMode) {
+        for (int i = 0; i < 5; ++i) {
+            Submonitor_Client_List->at(i).SendMessage((char *)"$KILL_SOCKET", 13);  // Terminate program
+        }
+    }
+    exit(signum);
+}
+
+mainmonitor::mainmonitor() : monitor("mainmonitor") {
+    poll_server = new Server;
+    Submonitor_Client_List = &Submonitor_Clients;
+}
+mainmonitor::~mainmonitor() {
+    if (this->GetDummyMode() && poll_server) {
+        delete[] poll_server;
+    }
+}
+void mainmonitor::CloseSubmonitors() {
+    if (!GetDummyMode()) {
+        for (int i = 0; i < GetMaxNumSubMonitors(); ++i) {
+            GetSubClientVec()->at(i).Close();
+        }
+    }
+}
+void mainmonitor::RelayPoll2msg(char *buffer, size_t msg_size) {
+    for (int i = 0; i < MAX_NUM_SUBMONITORS; ++i) {
+        Submonitor_Clients.at(i).SendMessage(buffer, msg_size);
+    }
+}
+
+void mainmonitor::OpenSubmonitorSockets() {
+    std::cout << " Opening relay ports for submonitors" << std::endl;
+    std::string submonitorPortMessage = " Opened Client on ports ";
+    for (int i = 0; i < MAX_NUM_SUBMONITORS; ++i) {
+        Submonitor_Clients.emplace_back(Client());
```

Comment:
Since Client has a parameterless constructor, using emplace_back(Client()) here implicitly uses a copy constructor.

Reply:
emplace_back only has move call signs, the destructor will be called on the r-value temporary when it gets moved into the vector. 
All will be fine since how little the monitor program actually does and since we limit the number of submonitors statically within the source code and not passed in as a command line parameter the accidental 6 extra allocations will be fine.
This does make it do extra work, but the vector takes ownership of the object after the emplace_back.
When the vector is destroyed at close of program the memory will be freed still.

Reply:
I just wanted to point this out, it's not an issue for the functionality, just a nitpick on my part.